TrafficSimulator/Assets/User/UserSelectManager.cs

```diff
@@ -0,0 +1,193 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace User
+{
+    /// <summary>
+    ///     Manages user selection of objects in the scene.
+    /// </summary>
+    public class UserSelectManager : MonoBehaviour
+    {
+        /// <summary>
+        ///     Delegate for handling double-clicked selected game object.
+        /// </summary>
+        /// <param name="newSelectedGameObject">The new selected game object.</param>
+        public delegate void DoubleClickedSelectedGameObjectHandler(Selectable newSelectedGameObject);
+
+        /// <summary>
+        ///     Delegate for handling single-clicked selected game object.
+        /// </summary>
+        /// <param name="newSelectedGameObject">The new selected game object.</param>
+        public delegate void SelectedGameObjectChangedHandler(Selectable newSelectedGameObject);
+
+        private static UserSelectManager _instance;
+
+        [HideInInspector] public bool CanSelectNewObject;
+
+        private Camera _mainCamera;
+        private InputAction _clickInput;
+        private InputAction _doubleClickInput;
+        private bool _hasSelectedGameObject;
+        private InputAction _pointInput;
+
+        /// <summary>
+        ///     The currently selected game object.
+        /// </summary>
+        public Selectable SelectedGameObject { get; private set; }
+        
+        /// <summary>
+        ///     Singleton instance of the UserSelectManager.
+        /// </summary>
+        public static UserSelectManager Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = FindObjectOfType<UserSelectManager>();
+                    if (_instance == null)
+                    {
+                        var obj = new GameObject();
```

Comment:
```suggestion
                        GameObject obj = new GameObject();
```

Reply:
Yes, forgot to change my Rider auto formatting :)